test/queue_list_test.dart

```diff
@@ -260,20 +260,36 @@ void main() {
 
   test("cast does not throw on mutation when the type is valid", () {
     var patternQueue = new QueueList<Pattern>()..addAll(['a', 'b']);
-    QueueList<Object> untypedQueue = patternQueue;
-    var stringQueue = untypedQueue.cast<String>();
+    var stringQueue = patternQueue.cast<String>();
     stringQueue.addAll(['c', 'd']);
+    expect(
+      stringQueue,
+      const isInstanceOf<QueueList<Pattern>>(),
```

Comment:
Check for `QueueList<String>` here. That's the type you are casting to, and a type that `patternQueue` *doesn't* have, so it checks that something did happen.
(You could throw in a check that `stringQueue` and `patternQueue` are not *identical*, but it's not important after checking that `stringQueue` has a type that `patternQueue` doesn't).

This is the `expect` that needs the `skip` parameter, because in Dart 1 mode, the type check won't be true since `cast` returns `patternQueue` unmodified. The two tests below will still succeed when `stringQueue` and `patternQueue` are the same object.

Reply:
Done. Thanks @lrhn!